cms/djangoapps/contentstore/views/tests/test_item.py

```diff
@@ -2169,3 +2171,32 @@ def test_locked_unit_staff_only_message(self):
         self._verify_has_staff_only_message(xblock_info, True)
         self._verify_has_staff_only_message(xblock_info, True, path=self.FIRST_SUBSECTION_PATH)
         self._verify_has_staff_only_message(xblock_info, True, path=self.FIRST_UNIT_PATH)
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_self_paced_item_visibility_state(self, store_type):
+        """
+        Test that in self-paced course, item has `live` visibility state.
+        Test that when item was initially in `scheduled` state in instructor mode, change course pacing to self-paced,
+        now in self-paced course, item should have `live` visibility state.
+        """
+        SelfPacedConfiguration(enabled=True).save()
+
+        with self.store.default_store(store_type):
+            # Create course, chapter and setup future release date to make chapter in scheduled state
+            course = CourseFactory.create()
+            chapter = self._create_child(course, 'chapter', "Test Chapter")
+            self._set_release_date(chapter.location, datetime.now(UTC) + timedelta(days=1))
+
+            # Check that has scheduled state
+            xblock_info = self._get_xblock_info(chapter.location)
+            self._verify_visibility_state(xblock_info, VisibilityState.ready)
```

Comment:
@awaisdar001 `xblock_info` will only have one state at a time. So it would either have `ready` state or `live` visibility state. So it would be redundant to check for live state if we already have checked for ready state.